src/Microsoft.AspNetCore.Mvc.Testing/WebApplicationTestFixture.cs

```diff
@@ -12,20 +14,30 @@ namespace Microsoft.AspNetCore.Mvc.Testing
     /// <summary>
     /// Fixture for bootstrapping an application in memory for functional end to end tests.
     /// </summary>
+    /// <typeparam name="TTestClass">The test class. We use this class assembly to search for
+    /// assembly metadata attributes containing the content root paths to the web applications
+    /// with startup class <typeparamref name="TStartup"/>.</typeparam>
     /// <typeparam name="TStartup">The applications startup class.</typeparam>
-    public class WebApplicationTestFixture<TStartup> : IDisposable where TStartup : class
+    public class WebApplicationTestFixture<TTestClass, TStartup> : IDisposable where TTestClass : class where TStartup : class
```

Comment:
This is kinda a big change. Most people aren't used to seeing CRTP

Reply:
If we really need to find the test class, it might be a better choice to provide a base class in place of this thing.

Reply:
I've solved the need for a generic in an acceptable way. I use the DependencyContext to find out the test assembly, which is something that we can do given that the host runs testHost in the context of the application and we allready require you to preserve the compilation context.